dts/bindings/stepper_motor/stepper-motor-device.yaml

```diff
@@ -0,0 +1,58 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+# SPDX-License-Identifier: Apache-2.0
+
+description: Stepper Motor Device
+
+include: base.yaml
+
+on-bus: motor
+
+compatible: "stepper-motor-device"
+
+properties:
+
+  friendly-name:
+    type: string
+    description: |
+      Human readable string describing the stepper motor. It can be used to
+      distinguish multiple instances of the same model.
+
+  gear-ratio:
+    type: string
+    default: "1.0"
+    description: |
+      Gear ratio from stepper motor to end transmission, This value can be in float and hence
+      the property type string
```

Comment:
please avoid pseudo floating points in DT, just specify the ratio with numerator/denominator, see eg voltage divider

Reply:
But why? 

With [DT_STRING_UNQUOTED()](https://docs.zephyrproject.org/apidoc/latest/group__devicetree-generic-prop.html#gad71ae303ce20e116a75c23ca552c2225) you can get a string property's value as an unquoted sequence of tokens.

It removes "the quotes" from string-valued properties. That can be useful, for example, when defining floating point values as a string in devicetree that you would like to use to initialize a float or double variable in C.